packages/components/overflow-items/overflow-items.ts

```diff
@@ -196,96 +195,76 @@ export class KbqOverflowItems {
      */
-    private updateItemsVisibility(): void {
-        const itemElementRefs = this.itemElementRefs();
-        const items = this.items().map(({ hidden }, index, _items) => {
-            return {
-                hidden,
-                isFirst: index === 0,
-                isLast: index === _items.length - 1,
-                elementRef: itemElementRefs[index]
-            };
+    private updateItemsVisibility(
+        items: readonly KbqOverflowItem[],
+        itemElementRefs: readonly ElementRef<HTMLElement>[],
+        reverseOverflowOrder: boolean,
+        resultElementRef: ElementRef<HTMLElement> | undefined,
+        { offsetWidth: totalWidth }: HTMLElement
+    ): void {
+        items.forEach((item) => {
+            item.hidden = false;
         });
-        const reverseOverflowOrder = this.reverseOverflowOrder();
-        const isEdgeItem = ({ isFirst, isLast }: (typeof items)[number]) => (reverseOverflowOrder ? isFirst : isLast);
-        const { offsetWidth: totalWidth, scrollWidth } = this.elementRef.nativeElement;
-        const isOverflown = scrollWidth > totalWidth;
-        if (isOverflown) {
-            const isVisibleItem = ({ hidden }: (typeof items)[number]) => !hidden();
-            const lastVisibleItem = !reverseOverflowOrder
-                ? this.findLast(items, isVisibleItem)
-                : items.find(isVisibleItem);
-            if (lastVisibleItem) {
-                lastVisibleItem.hidden.set(true);
-                if (!isEdgeItem(lastVisibleItem)) {
-                    this.updateItemsVisibility();
-                }
-            }
-        } else {
-            const isHiddenItem = ({ hidden }: (typeof items)[number]) => hidden();
-            const firstHiddenItem = !reverseOverflowOrder
-                ? items.find(isHiddenItem)
-                : this.findLast(items, isHiddenItem);
-            if (firstHiddenItem) {
-                const firstHiddenItemWidth = firstHiddenItem.elementRef.nativeElement.offsetWidth;
-                const visibleItemsTotalWidth = items
-                    .filter(({ hidden }) => !hidden())
-                    .reduce((width, { elementRef }) => width + elementRef.nativeElement.offsetWidth, 0);
-                const _isEdgeItem = isEdgeItem(firstHiddenItem);
-                const resultWidth = _isEdgeItem ? 0 : this.resultElementRef()?.nativeElement.offsetWidth || 0;
-                if (firstHiddenItemWidth + visibleItemsTotalWidth + resultWidth <= totalWidth) {
-                    firstHiddenItem.hidden.set(false);
-                    if (!_isEdgeItem) {
-                        this.updateItemsVisibility();
-                    }
+        let itemsWidth = items.reduce((width, _, index) => width + itemElementRefs[index].nativeElement.offsetWidth, 0);
+        const startIndex = reverseOverflowOrder ? 0 : items.length - 1;
+        const endIndex = reverseOverflowOrder ? items.length : -1;
```

Comment:
не так разве должно быть?

```suggestion
        const endIndex = reverseOverflowOrder ? items.length - 1 : 0;
```

Reply:
нет, тогда первый элемент не скроется
```ts
for (...; index !== endIndex; ...) {
```